drivers/imx-enet.c

```diff
@@ -513,9 +513,11 @@ static void enet_irq_thread(void *arg)
 {
 	enet_state_t *state = arg;
 	size_t rx_done = 0;
+	unsigned exit;
 
 	mutexLock(state->irq_lock);
-	while (state->drv_exit == 0) {
+	exit = atomic_load(&state->drv_exit);
+	while ((exit & EV_BUS_ERROR) == 0) {
```

Comment:
why checking only one bit? 
I don't care that currently only this bit is being set, looking at this code I would assume there are other non-fatal errors (incorrectly passed by `drv_exit` variable?)